diff --git a/09-countOfAllOddNumbers/countOfAllOddNumbers.js b/09-countOfAllOddNumbers/countOfAllOddNumbers.js
index 0f3ce57..bf1a73e 100644
--- a/09-countOfAllOddNumbers/countOfAllOddNumbers.js
+++ b/09-countOfAllOddNumbers/countOfAllOddNumbers.js
@@ -1,5 +1,13 @@
 function countOfAllOddNumbers(nums) {
-	// Insert code here;
+	let oddNumberCount = 0;
+
+	for (const number of nums) {
+		if (number % 2 !== 0) {
+			oddNumberCount++;
+		}
+	}
+
+	return oddNumberCount;
 }
 
 // Do not edit this line;
