diff --git a/10-countOfNumsWithinARange/countOfNumsWithinARange.js b/10-countOfNumsWithinARange/countOfNumsWithinARange.js
index 1f0f06c..f446afe 100644
--- a/10-countOfNumsWithinARange/countOfNumsWithinARange.js
+++ b/10-countOfNumsWithinARange/countOfNumsWithinARange.js
@@ -1,5 +1,14 @@
 function countOfNumsWithinARange(nums, start, end) {
-	// Insert code here;
+	let numsInRangeCount = 0;
+
+	for (const number of nums) {
+		// increment count if number is in range from start to end, both inclusive
+		if (number >= start && number <= end) {
+			numsInRangeCount++;
+		}
+	}
+
+	return numsInRangeCount;
 }
 
 // Do not edit this line;
