diff --git a/12-fizzBuzz/fizzBuzz.js b/12-fizzBuzz/fizzBuzz.js
index 9027016..5b1dce0 100644
--- a/12-fizzBuzz/fizzBuzz.js
+++ b/12-fizzBuzz/fizzBuzz.js
@@ -1,5 +1,23 @@
 function fizzBuzz(start, end) {
-  // Insert code here;
+  let arr = [];
+
+  // include start value and end value
+  for (let i = start; i <= end; i++) {
+    let result = i;
+  
+    // check for both conditions first, then individual conditions
+    if (i % 3 === 0 && i % 5 === 0) {
+      result = "FizzBuzz";
+    } else if (i % 3 === 0) {
+      result = "Fizz";
+    } else if (i % 5 === 0) {
+      result = "Buzz";
+    } 
+    
+    arr.push(result);
+  }
+
+  return arr;
 }
 
 // Do not edit this line;
