diff --git a/05-countOfAllBooleansAndStrings/countOfAllBooleansAndStrings.js b/05-countOfAllBooleansAndStrings/countOfAllBooleansAndStrings.js
index d59e21a..9ced2da 100644
--- a/05-countOfAllBooleansAndStrings/countOfAllBooleansAndStrings.js
+++ b/05-countOfAllBooleansAndStrings/countOfAllBooleansAndStrings.js
@@ -1,5 +1,14 @@
 function countOfAllBooleansAndStrings(arr) {
-  // Insert code here;
+	let booleanAndStringCount = 0;
+
+	for (const element of arr) {
+		// typeof checks the element's data type
+		if (element === true || element === false || typeof element === "string") {
+			booleanAndStringCount++;
+		}
+	}
+
+	return booleanAndStringCount;
 }
 
 // Do not edit this line;
