diff --git a/11-sumOfMinimumAndMaximum/sumOfMinimumAndMaximum.js b/11-sumOfMinimumAndMaximum/sumOfMinimumAndMaximum.js
index 3c0ff9b..b4bb88d 100644
--- a/11-sumOfMinimumAndMaximum/sumOfMinimumAndMaximum.js
+++ b/11-sumOfMinimumAndMaximum/sumOfMinimumAndMaximum.js
@@ -1,5 +1,19 @@
 function sumOfMinimumAndMaximum(nums) {
-  // Insert code here;
+  // assuming the array nums is not sorted, 
+  // need to loop through entire array
+
+  let minNum = nums[0];
+  let maxNum = nums[0];
+
+  for (let i = 1; i < nums.length; i++) {
+    if (nums[i] < minNum) {
+      minNum = nums[i];
+    } else if (nums[i] > maxNum) {
+      maxNum = nums[i];
+    }
+  }
+  
+  return minNum + maxNum;
 }
 
 // Do not edit this line;
