diff --git a/17-setUnionOfAnyAmountOfSets/setUnionOfAnyAmountOfSets.js b/17-setUnionOfAnyAmountOfSets/setUnionOfAnyAmountOfSets.js
index bac6a2a..94a91ce 100644
--- a/17-setUnionOfAnyAmountOfSets/setUnionOfAnyAmountOfSets.js
+++ b/17-setUnionOfAnyAmountOfSets/setUnionOfAnyAmountOfSets.js
@@ -1,5 +1,13 @@
 function setUnionOfAnyAmountOfSets(...args) {
-  // Insert code here;
+  let unionSet = new Set();
+  for (const set of args) {
+    for (const number of set) {
+      // 'Set' add() method will not add elements already in the set
+      unionSet.add(number);
+    }
+  }
+
+  return unionSet;
 }
 
 // Do not edit this line;
