diff --git a/14-frequencyCounter/frequencyCounter.js b/14-frequencyCounter/frequencyCounter.js
index 99fa3a6..15bf4df 100644
--- a/14-frequencyCounter/frequencyCounter.js
+++ b/14-frequencyCounter/frequencyCounter.js
@@ -1,5 +1,14 @@
 function frequencyCounter(word) {
-  // Insert code here;
+  let freqTracker = {};
+
+  for (const letter of word) {
+
+    // if letter already present, increment frequency, else add new letter with freq 1
+    freqTracker.hasOwnProperty(letter) ? freqTracker[letter]++ : freqTracker[letter] = 1;
+    console.log(freqTracker);
+  }
+
+  return freqTracker;
 }
 
 // Do not edit this line;
